core/src/main/java/io/grpc/transport/AbstractServerStream.java

```diff
@@ -81,9 +82,19 @@ protected ServerStreamListener listener() {
   }
 
   @Override
-  protected void receiveMessage(InputStream is) {
+  protected void receiveMessage(Iterator<InputStream> messages) {
+    // Discussion point. If a stream is closed but someone calls 'request' on it what should the
+    // behavior be. No more messages will come from the transport so throwing seems unnecessarily
+    // aggressive
+    if (inboundPhase() == Phase.STATUS) {
+      return;
```

Comment:
@louiscryan I'm guessing this is OBE at this point?  Probably the messageProducer's responsibility to close any remaining messages at shutdown, correct?